### PR TITLE
[Mellanox] fix the issue that failing to test whether dom supported prior to reading dom data

### PR DESCRIPTION
**Pull request by stephenxs**
**- What I did**
 fix the issue that failing to test whether dom supported prior to reading dom data.
it depends on pr [https://github.com/Azure/sonic-buildimage/pull/3118)](https://github.com/Azure/sonic-buildimage/pull/3118)

**- How I did it**
test dom support prior to reading dom data for SFP.

**- How to verify it**
check whether log "Cannot get Module EEPROM data: Invalid argument" can be found in syslog in a testbed that has some SFP modules without DOM supporting.

**- Description for the changelog**
 fix the issue that failing to test whether dom supported prior to reading dom data

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 1]
```diff
@@ -10,6 +10,9 @@
 except ImportError as e:
     raise ImportError("%s - required module not found" % str(e))
 
+#sfp supports dom
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: @stephenxs please add extra space:
```
# sfp supports dom
```

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 2]
```diff
@@ -517,10 +520,15 @@ def get_transceiver_dom_info_dict(self, port_num):
             offset = 256
 
             eeprom_raw = ['0'] * 256
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: @stephenxs please add named definition for '0' and 256

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 3]
```diff
@@ -517,10 +520,15 @@ def get_transceiver_dom_info_dict(self, port_num):
...
+            eeprom_raw[XCVR_DOM_CAPABILITY_OFFSET : XCVR_DOM_CAPABILITY_OFFSET + XCVR_DOM_CAPABILITY_WIDTH] = \
+                self._read_eeprom_specific_bytes_via_ethtool(port_num, XCVR_DOM_CAPABILITY_OFFSET, XCVR_DOM_CAPABILITY_WIDTH)
             sfp_obj = sff8472InterfaceId()
             calibration_type = sfp_obj._get_calibration_type(eeprom_raw)
 
+            dom_supported = (int(eeprom_raw[XCVR_DOM_CAPABILITY_OFFSET], 16) & XCVR_DOM_CAPABILITY_DOM_SUPPORT_BIT != 0)
+            if not dom_supported:
+                return transceiver_dom_info_dict
+
             eeprom_domraw = self._read_eeprom_specific_bytes_via_ethtool(port_num, offset, 256)
```
Comment (nazariig) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: @stephenxs please add named definition for 256

**Comment by lguohan** [comment]
retest this please

**Pull request merged**